src/plugins/discover/public/application/main/components/top_nav/open_alerts_popover.tsx

```diff
@@ -0,0 +1,184 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useState, useMemo } from 'react';
+import ReactDOM from 'react-dom';
+import { I18nStart } from 'kibana/public';
+import { EuiWrappingPopover, EuiLink, EuiContextMenu, EuiToolTip } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ISearchSource } from '../../../../../../data/common';
+import { KibanaContextProvider } from '../../../../../../kibana_react/public';
+import { DiscoverServices } from '../../../../build_services';
+import { updateSearchSource } from '../../utils/update_search_source';
+import { useDiscoverServices } from '../../../../utils/use_discover_services';
+
+const container = document.createElement('div');
+let isOpen = false;
+
+const ALERT_TYPE_ID = '.es-query';
+
+interface AlertsPopoverProps {
+  onClose: () => void;
+  anchorElement: HTMLElement;
+  searchSource: ISearchSource;
+}
+
+export function AlertsPopover(props: AlertsPopoverProps) {
+  const dataView = props.searchSource.getField('index')!;
+  const searchSource = props.searchSource;
+  const services = useDiscoverServices();
+  const { triggersActionsUi } = services;
+  const [alertFlyoutVisible, setAlertFlyoutVisibility] = useState(false);
+
+  const onCloseAlertFlyout = useCallback(() => props.onClose(), [props]);
+
+  /**
+   * Provides the default parameters used to initialize the new rule
+   */
+  const getParams = useCallback(() => {
+    const nextSearchSource = searchSource.createCopy();
+    updateSearchSource(nextSearchSource, true, {
+      indexPattern: searchSource.getField('index')!,
+      services,
+      sort: [],
+      useNewFieldsApi: true,
+    });
+
+    return {
+      searchType: 'searchSource',
+      searchConfiguration: nextSearchSource.getSerializedFields(),
+    };
+  }, [searchSource, services]);
+
+  const SearchThresholdAlertFlyout = useMemo(() => {
+    if (!alertFlyoutVisible) {
+      return;
+    }
+    return triggersActionsUi?.getAddAlertFlyout({
+      consumer: 'discover',
+      onClose: onCloseAlertFlyout,
+      canChangeTrigger: false,
+      ruleTypeId: ALERT_TYPE_ID,
+      initialValues: {
+        params: getParams(),
+      },
+    });
+  }, [getParams, onCloseAlertFlyout, triggersActionsUi, alertFlyoutVisible]);
+
+  const hasTimeFieldName = dataView.timeFieldName;
+  let createSearchThresholdRuleLink = (
+    <EuiLink onClick={() => setAlertFlyoutVisibility(true)} disabled={!hasTimeFieldName}>
+      <FormattedMessage
+        id="discover.alerts.createSearchThreshold"
+        defaultMessage="Create search threshold rule"
+      />
+    </EuiLink>
+  );
+
+  if (!hasTimeFieldName) {
+    const toolTipContent = (
+      <FormattedMessage
+        id="discover.alerts.missedTimeFieldToolTip"
+        defaultMessage="Data view does not have a time field."
+      />
+    );
+    createSearchThresholdRuleLink = (
```

Comment:
nit: can this be initialized in one go on line 74?

Reply:
Not really, because we need tool tip wrapper to explain disabled button. Otherwise tooltip is redundant.

![D02E160D-BEEC-4D0F-83B2-14F5BCEE84CC](https://user-images.githubusercontent.com/39378793/159241780-1a820b43-5e8a-4d49-9caa-f617c0a5727e.jpeg)